service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gceinstance/CreateGceInstanceStep.java

```diff
@@ -0,0 +1,296 @@
+                .setAutoDelete(true)
+                .setInitializeParams(
+                    new AttachedDiskInitializeParams()
+                        .setSourceImage(creationParameters.getVmImage())
+                        .setDiskSizeGb(Long.valueOf(100)))));
+
+    instance.setServiceAccounts(
+        List.of(
+            new ServiceAccount().setEmail(serviceAccountEmail).setScopes(SERVICE_ACCOUNT_SCOPES)));
+
+    List<ApiGcpGceInstanceGuestAccelerator> guestAccelerators =
+        creationParameters.getGuestAccelerators();
+    if (guestAccelerators != null) {
+      instance.setGuestAccelerators(
+          guestAccelerators.stream()
+              .map(
+                  accelerator ->
+                      new AcceleratorConfig()
+                          .setAcceleratorType(accelerator.getType())
+                          .setAcceleratorCount(accelerator.getCardCount()))
+              .collect(Collectors.toList()));
+    }
+    // Set metadata
+    Map<String, String> metadata = new HashMap<>();
+    Optional.ofNullable(creationParameters.getMetadata()).ifPresent(metadata::putAll);
+    addDefaultMetadata(metadata, workspaceUserFacingId, cliServer);
+    instance.setMetadata(
+        new Metadata()
+            .setItems(
+                metadata.entrySet().stream()
+                    .map(i -> new Items().setKey(i.getKey()).setValue(i.getValue()))
+                    .collect(Collectors.toList())));
+    return instance;
+  }
+
+  private static void addDefaultMetadata(
+      Map<String, String> metadata, String workspaceUserFacingId, String cliServer) {
+    if (metadata.containsKey(WORKSPACE_ID_METADATA_KEY)
+        || metadata.containsKey(SERVER_ID_METADATA_KEY)) {
+      throw new ReservedMetadataKeyException(
+          String.format(
+              "The metadata keys %s and %s are reserved for Terra.",
+              WORKSPACE_ID_METADATA_KEY, SERVER_ID_METADATA_KEY));
+    }
+    metadata.put(WORKSPACE_ID_METADATA_KEY, workspaceUserFacingId);
+    if (!StringUtils.isEmpty(cliServer)) {
+      metadata.put(SERVER_ID_METADATA_KEY, cliServer);
+    }
+  }
+
+  private static void setNetworks(Instance instance, String projectId, FlightMap workingMap) {
+    String region = workingMap.get(CREATE_RESOURCE_REGION, String.class);
+    String networkName = workingMap.get(CREATE_GCE_INSTANCE_NETWORK_NAME, String.class);
+    String subnetworkName = workingMap.get(CREATE_GCE_INSTANCE_SUBNETWORK_NAME, String.class);
+    instance.setNetworkInterfaces(
+        List.of(
+            new NetworkInterface()
+                .setNetwork(String.format("projects/%s/global/networks/%s", projectId, networkName))
+                .setSubnetwork(
+                    String.format(
+                        "projects/%s/regions/%s/subnetworks/%s", projectId, region, subnetworkName))
+                .setAccessConfigs(
+                    List.of(new AccessConfig().setType("ONE_TO_ONE_NAT").setName("External IP")))));
```

Comment:
extract private static final field for ONE_TO_ONE_NAT and add a description. same for External IP.